packages/utils/src/dom/get-element-by-url.ts

```diff
@@ -30,15 +30,28 @@ const getSrcsetUrls = (srcset: string): string[] => {
  */
 export const getElementByUrl = (dom: HTMLDocument, url: string): HTMLElement | null => {
     // TODO: Cache dom.querySelectorAll?.
-    const elements = dom.querySelectorAll('[href],[src],[poster],[srcset]').filter((element: any) => {
+    const elements = dom.querySelectorAll('[href],[src],[poster],[srcset]');
```

Comment:
Could we maybe restrict the selector? Something like `a[href],link[href],img[src],video[poster],img[srcset],source[srcset]` This should substantially reduce the number of calls to `hasAttrib` (which seems to be the bottleneck).

Reply:
Yes, I like the notion of scoping the selector to include element names - that should definitely help a bit. The only catch is to make sure we don't miss any. 

Looks like we'd need to include `audio[src]`, `iframe[src]`, `script[src]`, `source[src]`, and `track[src]` as well (and possibly `frame[src]` if we really want to be inclusive, though that's deprecated). I think we'd be okay omitting `applet`, `embed`, `frame`, and `object`, though.

Reply:
Applied this change and some further optimizations (attached new profile and screenshot to the pull request description). Looks like all the changes combined lead to a ~4x improvement.

Reply:
> Looks like all the changes combined lead to a ~4x improvement.

😍